java/Cherepnin_Oleh/src/com/zoolatech/lecture7/tasks/_2/SyncCalculator.java

```diff
@@ -0,0 +1,25 @@
+package com.zoolatech.lecture7.tasks._2;
+
+public class SyncCalculator implements Calculator {
+    private volatile int value;
+
+    public void add(int addedValue) {
+        synchronized (this) {
```

Comment:
it would be simpler to define the whole method as synchronized (it will be the same, but you'll need to write less code)